worldedit-core/src/main/java/com/fastasyncworldedit/core/util/TextureUtil.java

```diff
@@ -352,17 +367,47 @@ public TextureUtil(File folder) throws FileNotFoundException {
             try {
                 LOGGER.info("Downloading asset jar from Mojang, please wait...");
                 new File(Fawe.platform().getDirectory() + "/" + Settings.settings().PATHS.TEXTURES + "/").mkdirs();
-                try (BufferedInputStream in = new BufferedInputStream(
-                        new URL("https://piston-data.mojang.com/v1/objects/c0898ec7c6a5a2eaa317770203a1554260699994/client.jar")
-                                .openStream());
-                     FileOutputStream fileOutputStream = new FileOutputStream(
-                             Fawe.platform().getDirectory() + "/" + Settings.settings().PATHS.TEXTURES + "/1.19.2.jar")) {
-                    byte[] dataBuffer = new byte[1024];
-                    int bytesRead;
-                    while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
-                        fileOutputStream.write(dataBuffer, 0, bytesRead);
+
+                try {
+                    VersionMetadata metadata = getLatestVersion();
+                    LOGGER.info("Latest release version is {}", metadata.version());
+                    HashedResource resource = getLatestClientJarUrl(metadata);
+
+                    String outPath = Path.of(
+                            Fawe.platform().getDirectory().getPath(),
+                            Settings.settings().PATHS.TEXTURES,
+                            metadata.version() + ".jar"
+                    ).toString();
+                    // Copy resource to local fs
+                    try (final InputStream stream = new URL(resource.resource()).openStream();
+                         final ReadableByteChannel inChannel = Channels.newChannel(stream);
+                         final FileOutputStream out = new FileOutputStream(outPath);
+                         final FileChannel outChannel = out.getChannel()) {
+                        outChannel.transferFrom(inChannel, 0, Long.MAX_VALUE);
```

Comment:
```suggestion
                        stream.transferTo(out);
```

That allows to also remove the Channel resources.

Reply:
The reason for the usage of Channel was the significant file size of the client jar. By using channel the file may be copied directly, without buffering the whole jar file (even in chunks) into the memory. 

> This method is potentially much more efficient than a simple loop that reads from the source channel and writes to this channel. Many operating systems can transfer bytes directly from the source channel into the filesystem cache without actually copying them.

Reply:
That mainly applies to two file channels on one system, and even then the proper stream implementations just delegate to channels.